lib/Server.js

```diff
@@ -1845,10 +1845,19 @@ class Server {
         headers = headers(req, res, this.middleware.context);
       }
 
-      // eslint-disable-next-line guard-for-in
-      for (const name in headers) {
-        res.setHeader(name, headers[name]);
+      const allHeaders = [];
+
+      if (typeof headers === "object" && !Array.isArray(headers)) {
```

Comment:
Simplify it, `!Array.isArray(headers)` enough

Reply:
```suggestion
      if (!Array.isArray(headers)) {
```